cmd/token.go

```diff
@@ -0,0 +1,191 @@
+var tokenCmd = &cobra.Command{
+	Use:   "token",
+	Short: "Attest and fetch an OIDC token from Google Attestation Verification Service.",
+	Long: `Gather attestation report and send it to Google Attestation Verification Service for an OIDC token.
+The OIDC token includes claims regarding the authentication of the user by the authorization server (Google IAM server) with the use of an OAuth client application(Google Cloud apps). Note that this command will only work on a GCE VM with confidential space image for now. And Confidential computing API needs to be enabled for your account to access Google Attestation Verification Service https://pantheon.corp.google.com/apis/api/confidentialcomputing.googleapis.com.
+--algo flag overrides the public key algorithm for attestation key. If not provided then by default rsa is used.
+`,
+	Args: cobra.NoArgs,
+	RunE: func(*cobra.Command, []string) error {
+		rwc, err := openTpm()
+		if err != nil {
+			return err
+		}
+		defer rwc.Close()
+
+		// Metadata Server (MDS). A GCP specific client.
+		mdsClient = metadata.NewClient(nil)
+
+		ctx := namespaces.WithNamespace(context.Background(), namespaces.Default)
+		// TODO: principalFetcher is copied from go-tpm-tools/launcher/container_runner.go, to be refactored
+		// Fetch GCP specific ID token with specific audience.
+		// See https://cloud.google.com/functions/docs/securing/authenticating#functions-bearer-token-example-go.
+		principalFetcher := func(audience string) ([][]byte, error) {
+			u := url.URL{
+				Path: "instance/service-accounts/default/identity",
+				RawQuery: url.Values{
+					"audience": {audience},
+					"format":   {"full"},
+				}.Encode(),
+			}
+			idToken, err := mdsClient.Get(u.String())
+			if err != nil {
+				return nil, fmt.Errorf("failed to get principal tokens: %w", err)
+			}
+			fmt.Fprintf(debugOutput(), "GCP ID token fetched is: %s\n", idToken)
+			tokens := [][]byte{[]byte(idToken)}
+			return tokens, nil
+		}
+
+		fmt.Fprintf(debugOutput(), "Attestation Address is set to %s\n", asAddress)
+
+		region, err := getRegion(mdsClient)
+		if err != nil {
+			return fmt.Errorf("failed to fetch Region from MDS, the tool is probably not running in a GCE VM: %v", err)
+		}
+
+		projectID, err := mdsClient.ProjectID()
+		if err != nil {
+			return fmt.Errorf("failed to retrieve ProjectID from MDS: %v", err)
+		}
+
+		verifierClient, err := getRESTClient(ctx, asAddress, projectID, region)
+		if err != nil {
+			return fmt.Errorf("failed to create REST verifier client: %v", err)
+		}
+
+		// Supports GCE VM. Hard code the AK type. Set GCE AK (EK signing) cert
+		var gceAK *client.Key
+		if keyAlgo == tpm2.AlgRSA {
+			gceAK, err = client.GceAttestationKeyRSA(rwc)
+		}
+		if keyAlgo == tpm2.AlgECC {
+			gceAK, err = client.GceAttestationKeyECC(rwc)
+		}
+		if err != nil {
+			return err
+		}
+		if gceAK.Cert() == nil {
+			return errors.New("failed to find gceAKCert on this VM: try creating a new VM or contacting support")
```

Comment:
Add the keyAlgo to the error message. Also I think we can remove the "contacting support part" as it's vague. We can add "Verify the VM has an EK cert using get-shielded-identity gcloud command" etc.